### PR TITLE
add support for  aws s3 account public access block

### PR DESCRIPTION
**Pull request by moadibfr**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | yes
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #...
| ❓ Documentation  | yes 

## Description

add support for  aws s3 account public access block

### CODE REVIEW

File: enumeration/remote/aws/repository/s3control_repository.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	return s.accountId
+}
+
+func (s *s3ControlRepository) DescribeAccountPublicAccessBlock() (*s3control.PublicAccessBlockConfiguration, error) {
+	cacheKey := "S3DescribeAccountPublicAccessBlock"
+	if v := s.cache.Get(cacheKey); v != nil {
+		return v.(*s3control.PublicAccessBlockConfiguration), nil
+	}
+	out, err := s.clientFactory.GetS3ControlClient(nil).GetPublicAccessBlock(&s3control.GetPublicAccessBlockInput{
+		AccountId: aws.String(s.accountId),
```
Comment (eliecharra) [inline enumeration/remote/aws/repository/s3control_repository.go:38]: Probably better if this repo is stateless and this accountId is an argument of that function WDYT ?

File: enumeration/resource/resource_types.go [hunk 2]
```diff
@@ -99,11 +99,11 @@ var supportedTypes = map[string]ResourceTypeMeta{
 	"aws_s3_bucket_metric":                  {},
 	"aws_s3_bucket_notification":            {},
 	"aws_s3_bucket_policy":                  {},
-	"aws_s3_bucket_public_access_block":     {},
-	"aws_security_group": {children: []ResourceType{
+	"aws_s3_bucket_public_access_block":     {}, "aws_security_group": {children: []ResourceType{
```
Comment (eliecharra) [inline enumeration/resource/resource_types.go:102]: Lint issues here

File: pkg/cmd/scan/output/html_test.go [hunk 3]
```diff
@@ -236,7 +236,7 @@ func TestHTML_Write(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			tempDir := t.TempDir()
-			tempFile, err := ioutil.TempFile(tempDir, "result")
+			tempFile, err := os.CreateTemp(tempDir, "result")
```
Comment (eliecharra) [inline pkg/cmd/scan/output/html_test.go:239]: Is this something related to this PR or just a refactoring ? Does not matter it's not a big deal, just want to be sure I'm not missing something here 🤔

**Review by eliecharra — CHANGES_REQUESTED** [review]

File: enumeration/remote/aws/repository/s3control_repository.go (hunk 1 above)
Comment (moadibfr) [reply enumeration/remote/aws/repository/s3control_repository.go:36]: it's not really stateless as the provider is linked to the same account but I updated the injection

File: enumeration/remote/aws/repository/s3control_repository.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	cache         cache.Cache
+}
+
+func NewS3ControlRepository(factory client.AwsClientFactoryInterface, c cache.Cache) *s3ControlRepository {
+	return &s3ControlRepository{
+		clientFactory: factory,
+		cache:         c,
+	}
+}
+func (s *s3ControlRepository) GetAccountID() string {
```
Comment (eliecharra) [inline enumeration/remote/aws/repository/s3control_repository.go:26]: Is this method still used ?

**Review by eliecharra — DISMISSED** [review]
LGTM, just check if we could drop the `GetAccountID` method from the repo maybe

File: enumeration/remote/aws/repository/s3control_repository.go (hunk 4 above)
Comment (moadibfr) [reply enumeration/remote/aws/repository/s3control_repository.go]: yeah sorry I pushed an half done thing for you to comment on where the accountid was injected. It's fixed now.

**Review by eliecharra — APPROVED** [review]

**Comment by JAleksandrov** [comment]
After update seems like it could no longer scan my state:

Scanned resources
NoSuchPublicAccessBlockConfiguration: The public access block configuration was not found
status code: 404

**Comment by moadibfr** [comment]
> After update seems like it could no longer scan my state:
> 
> Scanned resources NoSuchPublicAccessBlockConfiguration: The public access block configuration was not found status code: 404

thanks @JAleksandrov just reproduced that and working on it: https://github.com/snyk/driftctl/issues/1590
🙏
